src/main/java/com/google/sps/data/ShoppingCartHandler.java

```diff
@@ -0,0 +1,73 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
+
+import java.util.ArrayList;
+
+public class ShoppingCartHandler {
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  /**
+   * Return an array with the current list of books of the current user.
+   */
+
+  public ArrayList<String> getUserShoppingCart(String uid) {
+    Query query = new Query("UserCart").setFilter(new Query.FilterPredicate("uid", Query.FilterOperator.EQUAL, uid));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity = results.asSingleEntity();
+    ArrayList<String> books = new ArrayList<>();
```

Comment:
Ideally you would just return list, even if you initialize it as an arraylist.
https://www.java67.com/2016/01/difference-between-list-and-arraylist-variable-in-java.html

Reply:
Done